### ISSUE TITLE
[feature] support Chinese Pinyin input when fuzzy find

### ISSUE THREAD

**Issue by roachsinai** [junegunn/fzf#2227]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish
    
## What expected

For example [Pinyin](https://en.wikipedia.org/wiki/Pinyin) of 操作系统 (Operation System in Englist) is `cao zuo xi tong`:

```
hmanx ~ » pypinyin "操作系统"
cāo zuò xì tǒng
hmanx ~ » pypinyin "操作系统" -s NORMAL
cao zuo xi tong
hmanx ~ » pypinyin "曹坐系桶" -s NORMAL             
cao zuo xi tong
```

And as third example above different Chinese chars may have same pinyin. So when use fzf to find files which in Chinses, you input English letters then choose Chinese chars you want in option list then fzf find the matches.

[pypinyin](https://github.com/mozillazg/python-pinyin) use `-s` option can change pinyin to English letters (without tone). So there is a way to add a fzf option that force the files list processed by pypinyin enable input letters to match Chinese chars directly? What's more, what show in fzf window is still Chinese chars not pinyin.

Maybe this feature is also good to Janpanese, Korean or any language has a tool to transform their pronounce to alphabets.

**Comment by copent** [comment]
Support, maybe also 'czxt' for search of "操作系统".

**Comment by copent** [comment]
the go version of pinyin: https://github.com/mozillazg/go-pinyin

**Comment by roachsinai** [comment]
> Support, maybe also 'czxt' for search of "操作系统".

If this issue is suppot `czxt` will support absolutely, as tihs is fuzzy finder. 😃

**Comment by jazmit** [comment]
I agree with Jiehong, implementing an IM is definitely not what fzf's job is, however allowing fzf to be configurable enough to support this would be really useful for this (and other) use cases.

Really all we're asking for is the ability to configure an arbitary executable as a String=>String transformation in fzf's pipeline.  Is this possible?

The suggested solution of using an IME is suboptimal because you must first switch to the input method, then select the correct characters, then further select the matching file, or two additional steps before you can open your file.

**Comment by jazmit** [comment]
After reading through the help page, I noticed that fzf already normalizes latin characters, and this can be disabled with `--literal`, this feature is really just a request for another kind of normalization.

**Comment by junegunn** [comment]
I'm not going to implement this directly in fzf, but you can probably use `search` and `transform-search` actions to implement custom translation of query strings.
https://junegunn.github.io/fzf/releases/0.59.0/#search-and-transform-search-action
